app.rb

```diff
@@ -1,20 +1,135 @@
 require_relative "data"
-apartments = data[:apartments]
-tenants = data[:tenants]
+require_relative "models/apartment"
+require_relative "models/tenant"
+apartments_old = data[:apartments]
+tenants_old = data[:tenants]
+apartments = []
+tenants = []
+
+#id, address, monthly_rent, square_feet
+apartments_old.map do |unit|
+  unit_obj = Apartment.new(unit[:id], unit[:address], unit[:monthly_rent], unit[:square_feet])
+  apartments << unit_obj
+end
+
+tenants_old.map do |renter|
+  renter_obj = Tenant.new(renter[:id], renter[:name], renter[:age], renter [:apartment_id])
+  tenants << renter_obj
+end
+
+# puts apartments
+# puts tenants
+
 
 # First, Open the data.rb an inspect the data. Identify and write, in comments, the following:
   # Explain how the data is structured
+    # The data is split between two related layers, tenants and apartments.
+    # tenants are linked to apartment via their apparment_id property.
+    # An apartment has many tenants, each tenant has one apartment.
   # What are the properties for each of the two types of hashes
+    # Apartments have an id, address, square_feet, and monthly_rent
+    # Tenants have an id, age, name, and apartment_id
+# # Use enumerables to -
+#   # Print all the addresses for the apartments
+#   apartments_old.map do |unit|
```

Comment:
When your goal is to build a new array, use `map`. While this works, in this case your goal is the side effect of `puts`ing, so I'd use `each` instead.